shared/constants/searchv3.js

```diff
@@ -31,11 +37,183 @@ export type RowProps = {|
   rightUsername: ?string,
 
   showTrackerButton: boolean,
-
-  onShowTracker: () => void,
 |}
 
+// Keypaths - maybe these should be somewhere else?
+export type KeyPath = ['searchv3Chat'] | ['searchv3Profile']
+
+// Actions
+
+export type Search = NoErrorTypedAction<
+  'searchv3:search',
+  {term: string, service: ?SearchPlatform, keyPath: KeyPath}
+>
+
+export type OnShowTracker = NoErrorTypedAction<'searchv3:onShowTracker', {resultId: string, keyPath: KeyPath}>
+
+// Helper
+function serviceNameToSearchPlatform(serviceName: string): SearchPlatform {
+  return {
+    keybase: 'Keybase',
+    twitter: 'Twitter',
+    github: 'Github',
+    reddit: 'Reddit',
+    hackernews: 'Hackernews',
+    pgp: 'Pgp',
+    facebook: 'Facebook',
+  }[serviceName]
+}
+
+// Platform icons
+
+function platformToIcon(platform: SearchPlatform): IconType {
+  return {
+    Keybase: 'iconfont-identity-devices',
+    Twitter: 'iconfont-identity-twitter',
+    Github: 'iconfont-identity-github',
+    Reddit: 'iconfont-identity-reddit',
+    Hackernews: 'iconfont-identity-hn',
+    Pgp: 'iconfont-identity-pgp',
+    Facebook: 'iconfont-identity-facebook',
+  }[platform]
+}
+
+function platformToLogo24(platform: SearchPlatform): IconType {
+  return {
+    Keybase: 'icon-keybase-logo-24',
+    Twitter: 'icon-twitter-logo-24',
+    Github: 'icon-github-logo-24',
+    Reddit: 'icon-reddit-logo-24',
+    Hackernews: 'icon-hacker-news-logo-24',
+    Pgp: 'icon-pgp-key-24',
+    Facebook: 'icon-facebook-logo-24',
+  }[platform]
+}
+
+// Parse fns
+
+export type RawResult = {
+  score: number,
+  keybase: ?{
+    username: string,
+    uid: string,
+    picture_url: ?string,
+    full_name: ?string,
+    is_followee: boolean,
+  },
+  service: ?{
+    service_name: PlatformsExpandedType,
+    username: string,
+    picture_url: ?string,
+    bio: ?string,
+    location: ?string,
+    full_name: ?string,
+  },
+}
+
+function externalServiceToId(serviceName: string, serviceUsername: string): string {
+  return `${serviceName}-${serviceUsername}`
+}
+
+function _parseKeybaseRawResultToRow(
+  result: RawResult,
+  isFollowingOnKeybase: boolean,
+  showTrackerButton: boolean
```

Comment:
Why make this an argument? Can this value always be determined from the `result` passed in?